### PR TITLE
dEta and dPhi windows similar to TP2014 algorithm

### PR DESCRIPTION
**Pull request by dildick**
@rekovic I updated the L1TkMuon and L1TkGlbMuon producers with the dEta and dPhi windows that were used for the Upgrade Technical Proposal studies. In each class I added two functions (the same for both cases) that retrieve the `sigmaEta` and `sigmaPhi`. Those values were then obtained with [1] and divided by sqrt(12). I calculated those 3 sigmaEta values and 1 sigmaPhi value in the latest CMSSW_6_2_X_SLHC release. You will find those numbers in `sigmaEtaTP()` and `sigmaPhiTP()`. With these additions the rate for both algorithms should drop significantly. 

In addition I modified the python configuration so that both the TP-like algorithm is run as well as the one developed by Luca Cadamuro.

Supporting material:
- https://indico.cern.ch/event/803818/contributions/3346646/attachments/1808078/2951955/L1Checks.pdf
- https://indico.cern.ch/event/803818/contributions/3343229/attachments/1807985/2951701/20190307_L1ObjectV7p3_L1Meeting_YiChen_v4.pdf

[1]
https://github.com/cms-sw/cmssw/blob/CMSSW_6_2_X_SLHC/L1Trigger/L1ExtraFromDigis/src/L1MuonParticleExtendedProducer.cc#L312-L340

### CODE REVIEW

File: L1Trigger/L1TTrackMatch/python/L1TkMuonProducer_cfi.py [hunk 1]
```diff
@@ -49,3 +49,7 @@
 
 
 )
+
+L1TkMuonsDynamicWindows = L1TkMuons.Clone(emtfMatchAlgoVersion='DynamicWindows')
+
+L1TkMuons = cms.Sequence(L1TkMuons * L1TkMuonsDynamicWindows)
```
Comment (mcepeda) [inline L1Trigger/L1TTrackMatch/python/L1TkMuonProducer_cfi.py]: Rename this so it does not have the same name as the L1TkMuon object (eg: L1TkMuonSequence ?)

File: L1Trigger/L1TTrackMatch/python/L1TkMuonProducer_cfi.py [hunk 2]
```diff
@@ -49,3 +49,7 @@
 
 
 )
+
+L1TkMuonsDynamicWindows = L1TkMuons.Clone(emtfMatchAlgoVersion='DynamicWindows')
```
Comment (mcepeda) [inline L1Trigger/L1TTrackMatch/python/L1TkMuonProducer_cfi.py]: cmsRun complains at this (Clone vs clone). I used:
L1TkMuonsDynamicWindows = L1TkMuons.clone() 
L1TkMuonsDynamicWindows.emtfMatchAlgoVersion=cms.string('DynamicWindows')

**Comment by dildick** [comment]
@mcepeda I noticed this on Friday. The PR was updated just now.

**Comment by rekovic** [comment]
@dildick 
This PR is changing the existing TkMuon producer and the TkMuon configuration. 
Please do not introduce changes to the existing TkMuon which is used as a baseline. Instead, please make a PR with an alternative producer which has the changes you propose. 

Please make sure that the algorithm you propose is presented and blessed in the Muon Trigger Object Group. 
Attn konstantinos.vellidis@cern.ch, Jacobo.Konigsberg@cern.ch

**Comment by mcepeda** [comment]
hi @rekovic , @dildick  - since the changes are not big, maybe it is simpler to not add yet another  producer but rather to set it up so that the user can choose deta/dphi matching at the python configuration. The TP/DynamicWindow is already a configuration level change.

**Comment by dildick** [comment]
@rekovic I understand your concern. With the extra option I defined there is no difference in the baseline L1TkMuons algorithm. Instead L1TkMuonsTP is the algorithm that leans closest to the TP algorithm.

**Comment by rekovic** [comment]
@dildick Yes, this is good. Thanks.

**Comment by dildick** [comment]
@rekovic Can you merge this PR?

**Comment by dildick** [comment]
Thank you!

**Pull request merged**